src/connector/src/source/kafka/private_link.rs

```diff
@@ -211,16 +213,18 @@ fn is_kafka_connector(with_properties: &BTreeMap<String, String>) -> bool {
 }
 
 pub fn insert_privatelink_broker_rewrite_map(
-    properties: &mut BTreeMap<String, String>,
+    with_options: &mut BTreeMap<String, String>,
     svc: Option<&PrivateLinkService>,
     privatelink_endpoint: Option<String>,
 ) -> anyhow::Result<()> {
     let mut broker_rewrite_map = HashMap::new();
-    let servers = get_property_required(properties, kafka_props_broker_key(properties))?;
+    let servers = get_property_required(with_options, kafka_props_broker_key(with_options))?;
     let broker_addrs = servers.split(',').collect_vec();
-    let link_target_value = get_property_required(properties, PRIVATE_LINK_TARGETS_KEY)?;
+    let link_target_value = get_property_required(with_options, PRIVATE_LINK_TARGETS_KEY)?;
     let link_targets: Vec<AwsPrivateLinkItem> =
         serde_json::from_str(link_target_value.as_str()).map_err(|e| anyhow!(e))?;
+    // remove the private link targets from WITH options, as they are useless after we constructed the rewrite mapping
+    with_options.remove(PRIVATE_LINK_TARGETS_KEY);
```

Comment:
If we go this way, we do not need to add them in the `KafkaPrivateLinkCommon` at all. And they are not passed there. We only need one of these two solution.

Reply:
I guess only adding this line without any other changes in this PR is enough to fix the issue.

Reply:
We can remove `private_link_targets` and `private_link_endpoint` from `KafkaPrivateLinkCommon`, and I'd like to move privatelink related properties to a dedicated structure instead put it in `KafkaCommon`